clustermesh-apiserver/clustermesh/root.go

```diff
@@ -389,7 +451,15 @@ func startServer(
 	ctx := context.Background()
 	go synchronize(ctx, resources.CiliumIdentities, newIdentitySynchronizer(ctx, logger, cinfo, backend, factory, syncState.WaitForResource()))
 	go synchronize(ctx, resources.CiliumNodes, newNodeSynchronizer(ctx, logger, cinfo, backend, factory, syncState.WaitForResource()))
-	go synchronize(ctx, resources.CiliumSlimEndpoints, newEndpointSynchronizer(ctx, logger, cinfo, backend, factory, syncState.WaitForResource()))
+
+	if option.Config.EnableCiliumEndpointSlice {
+		logger.Info("Synchronizing endpoints using CiliumEndpointSlices")
+		go synchronize(ctx, resources.CiliumEndpointSlices, newEndpointSynchronizer(ctx, logger, cinfo, backend, factory, syncState.WaitForResource(), option.Config.EnableCiliumEndpointSlice))
```

Comment:
The setting should be retrieved from the proper configuration struct injected via hive, instead of the global variable.